cmd/kube-proxy/proxy.go

```diff
@@ -17,43 +17,33 @@ limitations under the License.
 package main
 
 import (
-	"fmt"
+	goflag "flag"
 	"os"
 
 	"k8s.io/apiserver/pkg/server/healthz"
-	"k8s.io/apiserver/pkg/util/flag"
+	utilflag "k8s.io/apiserver/pkg/util/flag"
 	"k8s.io/apiserver/pkg/util/logs"
 	"k8s.io/kubernetes/cmd/kube-proxy/app"
-	"k8s.io/kubernetes/cmd/kube-proxy/app/options"
 	_ "k8s.io/kubernetes/pkg/client/metrics/prometheus" // for client metric registration
 	_ "k8s.io/kubernetes/pkg/version/prometheus"        // for version metric registration
 	"k8s.io/kubernetes/pkg/version/verflag"
 
 	"github.com/spf13/pflag"
 )
 
-func init() {
-	healthz.DefaultHealthz()
-}
-
 func main() {
-	config := options.NewProxyConfig()
-	config.AddFlags(pflag.CommandLine)
-
-	flag.InitFlags()
 	logs.InitLogs()
 	defer logs.FlushLogs()
 
-	verflag.PrintAndExitIfRequested()
+	healthz.DefaultHealthz()
 
-	s, err := app.NewProxyServerDefault(config)
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "%v\n", err)
-		os.Exit(1)
-	}
+	command := app.NewProxyCommand()
+
+	pflag.CommandLine.SetNormalizeFunc(utilflag.WordSepNormalizeFunc)
```

Comment:
I remember now - our utility function to initialize flags is calling `pflag.Parse()` which doesn't work with Cobra commands (https://github.com/kubernetes/kubernetes/blob/82826ec2737870a0c7d5a5bcef5faffe0c11580f/staging/src/k8s.io/apiserver/pkg/util/flag/flags.go#L50). I've updated this PR to comment to that effect.

I would like to see all the components be slightly refactored to make use of the Cobra commands that currently exist; they are currently only used to generate docs. xref https://github.com/kubernetes/kubernetes/issues/34732